File: articles/linear-algebra-for-systems-modeling/articles/inverse-matrices-and-structural-recovery/julia/inverse_recovery.jl
# Inverse Matrices and Structural Recovery

A = [3 1; 2 4]
x_original = [2; 1]
b = A * x_original

x_recovered = inv(A) * b

println("A = ", A)
println("Original state x = ", x_original)
println("Observed output b = Ax = ", b)
println("Recovered state inv(A)b = ", x_recovered)

File: articles/linear-algebra-for-systems-modeling/articles/inverse-matrices-and-structural-recovery/r/inverse_recovery.R
# Inverse Matrices and Structural Recovery

A <- matrix(c(3, 2, 1, 4), nrow = 2)
x_original <- matrix(c(2, 1), nrow = 2)
b <- A %*% x_original

x_recovered <- solve(A) %*% b

print("A:")
print(A)
print("Original state:")
print(x_original)
print("Observed output b = Ax:")
print(b)
print("Recovered state solve(A) %*% b:")
print(x_recovered)
